gcc10 complains about strncopy truncating output before terminating nul

Using gcc version 10.2.1:

```
In function ‘mpool_strndup’,
    inlined from ‘mpool_strdup’ at ../auxil/paraglob/src/ahocorasick/mpool.c:196:12:
../auxil/paraglob/src/ahocorasick/mpool.c:174:9: warning: ‘strncpy’ output truncated before terminating nul copying as many bytes from a string as its length [-Wstringop-truncation]
  174 |         strncpy((char *)ret, str, n);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
../auxil/paraglob/src/ahocorasick/mpool.c: In function ‘mpool_strdup’:
../auxil/paraglob/src/ahocorasick/mpool.c:194:11: note: length computed here
  194 |     len = strlen(str);
      |           ^~~~~~~~~~~
```

--- a/src/ahocorasick/mpool.c
+++ b/src/ahocorasick/mpool.c
@@ -171,7 +171,7 @@ void *mpool_strndup (struct mpool *pool, const char *str, size_t n)
 
     if ((ret = mpool_malloc(pool, n+1)))
     {
-        strncpy((char *)ret, str, n);
+        memcpy(ret, str, n);
         ((char *)ret)[n] = '\0';
     }
 

